ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -103,14 +117,40 @@ void handleRestoration(final RestoreCommands restoreCommands) {
    * @param command The string containing the statement to be executed
    * @param commandId The ID to be used to track the status of the command
    */
+  @SuppressWarnings("unchecked")
   void handleStatement(
       final Command command,
       final CommandId commandId
   ) {
-    handleStatementWithTerminatedQueries(command,
-                                         commandId,
-                                         null,
-                                         false);
+    final String statementText = command.getStatement();
+    if (statementText.equalsIgnoreCase(TerminateCluster.TERMINATE_CLUSTER_STATEMENT_TEXT)) {
+      this.ksqlEngine.stopAcceptingStatemens();
```

Comment:
Moved it to a method. 
Creating `TerminateCluster` instance would add more code since we can directly instantiate the source list and type from the command.